apps/marketplace/pages/SellerEditFlowPage.js

```diff
@@ -0,0 +1,153 @@
+import LoadingIndicatorFullPage from 'shared/LoadingIndicatorFullPage/LoadingIndicatorFullPage'
+import { SellerEditFormReducer } from 'marketplace/reducers'
+
+const model = 'SellerEditFlowPage'
+const processResponse = response => {
+  // only accept data defined in the form reducer
+  const data = { ...SellerEditFormReducer }
+  if (response.data) {
+    Object.keys(response.data).map(property => {
+      if (Object.keys(SellerEditFormReducer).includes(property)) {
+        data[property] = response.data[property]
+      }
+      return true
+    })
+  }
+  return data
+}
+
+export class SellerEditFlowPage extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      loading: false,
+      flowIsDone: false
+    }
+
+    this.handleStageMount = this.handleStageMount.bind(this)
+    this.save = this.save.bind(this)
+  }
+
+  componentDidMount() {
+    if (this.props.match.params.supplierCode) {
+      this.getSellerData()
+    }
+  }
+
+  getSellerData() {
+    this.setState({
+      loading: true
+    })
+    this.props.loadInitialData(this.props.match.params.supplierCode).then(response => {
+      const data = processResponse(response)
+      this.props.changeFormModel(data)
+      this.setState({
+        loading: false
+      })
+    })
+  }
+
+  save() {
+    const data = { ...this.props[model] }
+    this.setState({
+      loading: true
+    })
+    return this.props.saveSeller(this.props.match.params.supplierCode, data.supplier).then(response => {
+      if (response.status === 200) {
+        this.props.changeFormModel(processResponse(response))
+        this.setState({
+          loading: false
+        })
+        return Promise.resolve(response)
+      }
+      this.setState({
+        loading: false
+      })
+      return Promise.reject(response.data.message)
+    })
+  }
+
+  handleStageMount() {
+    this.props.resetFormValidity()
+  }
+
+  render() {
+    if (this.props.errorMessage) {
+      let hasFocused = false
+      const setFocus = e => {
+        if (!hasFocused) {
+          hasFocused = true
+          e.focus()
+        }
+      }
+      return (
+        <ErrorBoxComponent
+          title="A problem occurred when loading the brief details"
+          errorMessage={this.props.errorMessage}
+          setFocus={setFocus}
+          form={{}}
+          invalidFields={[]}
+        />
+      )
+    }
+    const supplierCode = this.props.match.params.supplierCode
+
+    if (this.state.loading) {
+      return <LoadingIndicatorFullPage />
+    }
+
+    if (this.state.flowIsDone) {
+      return <Redirect to={`${rootPath}/seller-edit/${supplierCode}/completed`} push />
+    }
+
+    return (
+      <ProgressFlow
+        model={model}
+        basename={`${rootPath}/seller-edit/${supplierCode}`}
+        stages={SellerEditStages}
+        onStageMount={this.handleStageMount}
+        saveModel={this.save}
+        returnPath={`${rootPath}/brief/${supplierCode}/overview/specialist`}
```

Comment:
Is this return path right?  `/brief/${supplierCode}`?

Reply:
go to seller-dashboard.. but this will never be hit